apps/wakunode2/wakunode2_validator_signed.nim

```diff
@@ -29,9 +33,26 @@ proc msgHash*(pubSubTopic: string, msg: WakuMessage): array[32, byte] =
   ctx.update(pubsubTopic.toBytes())
   ctx.update(msg.payload)
   ctx.update(msg.contentTopic.toBytes())
+  ctx.update(msg.timestamp.uint64.toBytes(Endianness.littleEndian))
+  ctx.update(if msg.ephemeral: @[1.byte] else: @[0.byte])
 
   return ctx.finish()
 
+proc withinTimeWindow*(msg: WakuMessage): bool =
+  # Returns true if the message timestamp is:
+  # abs(now - msg.timestamp) < MessageWindowInSec
+  let ts = msg.timestamp
+  let now = getNanosecondTime(getTime().toUnixFloat())
+  let window = getNanosecondTime(MessageWindowInSec)
+
+  if now > ts:
```

Comment:
open for other ways.

cant do abs(now-ts) since im working with **u**int64 and if I cast to **s**igned, I will lose precision. So its the way I found.

Reply:
Maybe https://nim-lang.org/docs/times.html#abs%2CDuration from `times` can be used

Reply:
Sorry, mabe it is a completely stupid question, but would losing precision matter? Is the concern that someone could make the difference between TS and now overflow and thus the result of abs() be < window?

Can we check for overflow?

Reply:
ou `between` is indeed what I needed, but since in nwaku we generally use our own `Timestamp` (and not te `Duration`) I think this may be simpler. Less castings, etc.

> Sorry, mabe it is a completely stupid question, but would losing precision matter? Is the concern that someone could make the difference between TS and now overflow and thus the result of abs() be < window?

mm havent done the maths if matters or not. Difference between `ts` and `now` will never overflow, since it will be always a smaller value. But since both values are `uint64` I need to do `ts-now` or `now-ts` to ensure the result is not negative. Hence why i compare.

Mathematically is just `abs(now-ts)` but in practice thats not possible afaik without converting to int64 with a possible lose in precission.

Reply:
Wait, but `Timestamp` is an alias for `int64` not `uint64`, unless I'm missing something?

Reply:
ouch indeed. Totally assumed it was `uint64`, since otherwise it would be a waste of bytes (assuming negative timestamps doesnt make sense). my bad.

fixed, way easier da7814a4dd33f9b1120f419763f100bbf67c8339

some quick napkin math @vpavlin int64 overflow with `9223372036854775807`, since its nano seconds thats `9223372036 . 854775807` and `9223372036` in unix timestamp its `Fri Apr 11 2262 23:47:16 GMT+0000` so we are covered.

thanks all!